### PR TITLE
Nuzhy/[QUILL-1613]/Input (Input Group)(Text + button)

### PR DESCRIPTION
**Pull request by Nuzhy-Deriv**
- Input (Input Group) (Text + Button) Component added

image

- Fix: setting status icon for `success`, `error` statuses.

### CODE REVIEW

File: lib/components/Input/base/index.tsx [hunk 1]
```diff
@@ -18,13 +18,15 @@ export type Variants = "fill" | "outline";
...
 export type TextAlignments = "left" | "center";
+type ButtonPositions = "right" | "bottom";
 
 export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
     type?: Types;
     leftIcon?: ReactNode;
     rightIcon?: ReactNode;
     inputSize?: TMediumSizes;
     status?: Status;
+    readOnly?: boolean;
```
Comment (jia-deriv) [inline lib/components/Input/base/index.tsx:29]:
```suggestion
```
We can remove this. `HTMLInputElement` will have `readOnly`

**Review by jia-deriv — APPROVED** [review]

File: lib/components/Input/InputGroup/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { forwardRef, useEffect, useState } from "react";
+import Input, { InputProps } from "@components/Input/base";
+import { Button, TVariant } from "@components/Button";
```
Comment (prince-deriv) [inline lib/components/Input/InputGroup/Button/index.tsx:3]: Need to refactor this in other PR: 
```suggestion
import { Button, TButtonVariant } from "@components/Button";
```

File: lib/components/Input/base/index.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
     status?: Status;
     disabled?: boolean;
     variant?: Variants;
-    message?: string;
+    message?: ReactNode;
```
Comment (prince-deriv) [inline lib/components/Input/base/index.tsx:31]:
```suggestion
    message?: ReactNode | string;
```

Comment (Nuzhy-Deriv) [reply lib/components/Input/base/index.tsx:31]: I omitted the string since `ReactNode` represents `string` as well. is this necessary?

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**